googlemock/src/gmock_main.cc

```diff
@@ -32,6 +32,20 @@
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
 
+#ifdef ARDUINO
+void setup() {
+  int argc = 0;
+  char** argv = nullptr;
```

Comment:
Arduino doesn't have a main() - but testing::InitGoogleMock expects proper argv/argc values.
argc==0 and argv==nullptr are not valid.

Here is one example how to fake this:

 int argc = 1;
  const char *arg0 = "ANYNAME"; // A dummy name for the executable. Any name will do.
  char *argv0 = (char *)arg0;
  char **argv = &argv0;
  ::testing::InitGoogleMock(&argc, argv);